chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiUpdater.java

```diff
@@ -458,35 +458,35 @@ public void checkForDuplicates() throws SQLException, Exception
 				+ "where ReferenceDatabase_2_name.name = 'ChEBI' and identifier is not null\n" + "group by ReferenceEntity.identifier\n"
 				+ "having count(ReferenceMolecule.DB_ID) > 1;\n";
 
-		ResultSet duplicates = adaptor.executeQuery(findDuplicateReferenceMolecules, null);
-		duplicatesLog.info("# DB_ID\tCreator\tDuplicated Identifier\tReferenceMolecule");
-
-		// Should only be one, but API returns collection.
-		@SuppressWarnings("unchecked")
-		Collection<GKInstance> chebiDBInsts = (Collection<GKInstance>)adaptor.fetchInstanceByAttribute(ReactomeJavaConstants.ReferenceDatabase, ReactomeJavaConstants.name, "=", "ChEBI");
-		GKInstance chebiDBInst = chebiDBInsts.stream().findFirst().orElse(null);
-		
-		AttributeQueryRequest chebiAQR = adaptor.new AttributeQueryRequest(ReactomeJavaConstants.ReferenceMolecule, ReactomeJavaConstants.referenceDatabase, "=", chebiDBInst);
-		
-		while (duplicates.next())
+		try(ResultSet duplicates = adaptor.executeQuery(findDuplicateReferenceMolecules, null))
 		{
-			String identifier = duplicates.getString(1);
-			int numberOfDuplicates = duplicates.getInt(2);
-			this.duplicatesSB.append("\n** ReferenceMolecule with identifier " + identifier + " occurs " + numberOfDuplicates + " times:\n\n");
-			String operator = identifier == null ? "IS NULL" : "=";
+			duplicatesLog.info("# DB_ID\tCreator\tDuplicated Identifier\tReferenceMolecule");
+	
+			// Should only be one, but API returns collection.
+			@SuppressWarnings("unchecked")
+			Collection<GKInstance> chebiDBInsts = (Collection<GKInstance>)adaptor.fetchInstanceByAttribute(ReactomeJavaConstants.ReferenceDatabase, ReactomeJavaConstants.name, "=", "ChEBI");
+			GKInstance chebiDBInst = chebiDBInsts.stream().findFirst().orElse(null);
```

Comment:
Is it possible that chebiDBInsts could be null triggering an NPE?

Reply:
I suppose it _could_ happen if there are no ReferenceDatabase objects with the name "ChEBI", in which case I think it's perfectly acceptable for the application to crash with an NPE. The presence of the "ChEBI" ReferenceDatabase object is necessary for the rest of the application to run.

Reply:
A similar query is performed earlier in the program (near the beginning of `updateChebiReferenceMolecules`). I'll added an explicit RuntimeException _there_ to ensure the program terminates if there is no ChEBI RefDB object.